### DISCUSSION TITLE
Generating OpenAPI YAML from Pydantic models

### DISCUSSION THREAD

**Discussion by miohtama** [pydantic/pydantic#10552] [Question]
My use case is that I have an existing hand written and manually managed OpenAPI YAML files. I'd like to slowly introduce Pydantic to the system, eventually letting it take over fully.

I'd like to take baby steps first. The first thing I would need is to export Pydantic models and make them part of the YAML repository.

Are there any exporters that export Pydantic models as OpenAPI YAML?

- I found several importers (other way around)
- I found JSON exporters
- ... but I could not find YAML exporters

**Comment by Viicos** [comment]
You can use the `model_json_schema` method on your Pydantic models to export a JSON Schema. OpenAPI is only relevant in Rest API design, but is based on the JSON Schema spec. The outputted JSON can easily be converted to YAML.
